### PR TITLE
Add store builds to nuget packaging

### PR DESCRIPTION
**Pull request by tiagoshibata**
Build UWP Nugets of ORT and WinML and package it into the bundled Nuget package

### CODE REVIEW

**Review by smk2007 — APPROVED** [review]

File: onnxruntime/core/framework/provider_bridge_ort.cc [hunk 1]
```diff
@@ -658,23 +658,6 @@ std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Dnnl(i
   static ProviderLibrary library(LIBRARY_PREFIX "onnxruntime_providers_dnnl" LIBRARY_EXTENSION);
   if (!library.provider_)
     return nullptr;
-
-#if defined(_WIN32) && !defined(_OPENMP)
```
Comment (yuslepukhin) [inline onnxruntime/core/framework/provider_bridge_ort.cc:662]: >if defined(_WI [](start = 1, length = 14)

The PR lacks description and motivation. In particular, why this piece of code being removed?

**Review by yuslepukhin — CHANGES_REQUESTED** [review]
Please, add PR documentation and motivation to help the review.
Also respond to comments.

File: onnxruntime/core/framework/provider_bridge_ort.cc (hunk 1 above)
Comment (RyanUnderhill) [reply onnxruntime/core/framework/provider_bridge_ort.cc:662]: It's ok. This code was originally moved into the dnnl provider itself. My TensorRT change also modified this code and when I merged with master, github added it back in and I didn't notice. This change just fixes the incorrectly added back code. (they let me know about this already)

All good!

Comment (tiagoshibata) [reply onnxruntime/core/framework/provider_bridge_ort.cc:662]: This is part of #4738 which got reverted by Ryan by accident, as he mentioned. I had discussed it with him offline.

**Review by smk2007 — DISMISSED** [review]

File: csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.targets [hunk 2]
```diff
@@ -9,9 +9,11 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <WindowsAIBinary>$(MSBuildThisFileDirectory)..\..\runtimes\win-$(WindowsAI-Platform)\native\Microsoft.AI.MachineLearning.dll</WindowsAIBinary>
-    <OnnxRuntimeBinary>$(MSBuildThisFileDirectory)..\..\runtimes\win-$(WindowsAI-Platform)\native\onnxruntime.dll</OnnxRuntimeBinary>
-    <DirectMLBinary>$(MSBuildThisFileDirectory)..\..\runtimes\win-$(WindowsAI-Platform)\native\directml.dll</DirectMLBinary>
+    <RuntimesDirectory Condition="'$(OutputType)' == 'AppContainerExe'">$(MSBuildThisFileDirectory)..\..\runtimes\win-$(WindowsAI-Platform)\uap</RuntimesDirectory>
```
Comment (ryanlai2) [inline csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.targets:12]: >AppContainerExe [](start = 54, length = 15)

What's AppContainerExe?

File: tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml [hunk 3]
```diff
@@ -40,42 +40,74 @@ steps:
...
       ${{ if and(eq(parameters.BuildArch, 'x86'), eq(parameters.BuildForStore, 'false')) }}:
         arguments: '--build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --enable_onnx_tests --enable_wcos $(TelemetryOption) --use_dml --use_winml --cmake_generator "Visual Studio 16 2019" --update --config RelWithDebInfo --enable_lto --disable_rtti --x86'
       ${{ if and(eq(parameters.BuildArch, 'x86'), eq(parameters.BuildForStore, 'true')) }}:
-        arguments: '--build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --enable_onnx_tests --enable_wcos $(TelemetryOption) --use_dml --use_winml --cmake_generator "Visual Studio 16 2019" --update --config RelWithDebInfo --enable_lto --disable_rtti --x86 --enable_windows_store'
+        arguments: '--build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --skip_winml_tests --enable_wcos $(TelemetryOption) --use_winml --cmake_generator "Visual Studio 16 2019" --update --config RelWithDebInfo --enable_lto --disable_rtti --x86 --enable_windows_store --target winml_dll'
```
Comment (ryanlai2) [inline tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml:47]: >skip_winml_tests [](start = 102, length = 16)

Why skip winml tests?

**Review by ryanlai2 — DISMISSED** [review]
:shipit:

File: csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.targets (hunk 2 above)
Comment (tiagoshibata) [reply csharp/src/Microsoft.AI.MachineLearning.Interop/Microsoft.AI.MachineLearning.targets:12]: Executable built to run in an appcontainer (the store container isolation)

File: tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml (hunk 3 above)
Comment (tiagoshibata) [reply tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml:47]: They can't be built as-is (tests use some desktop specific APIs) or run directly (you have to create a packaged app to run in the appcontainer environment)

Comment (martinb35) [reply tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml:47]: and was removing use_dml intentional?

File: tools/ci_build/build.py [hunk 4]
```diff
@@ -176,6 +176,9 @@ def parse_arguments():
         "--cmake_extra_defines", nargs="+",
         help="Extra definitions to pass to CMake during build system "
         "generation. These are just CMake -D options without the leading -D.")
+    parser.add_argument(
+        "--target",
+        help="Build a specific target")
```
Comment (yuslepukhin) [inline tools/ci_build/build.py:181]: It would be nice to give an example of a target. I can not come with any

**Comment by yuslepukhin** [comment]
> std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Dnnl(int device_id) {

This file looks unchanged

---
Refers to: onnxruntime/core/framework/provider_bridge_ort.cc:673 in 89bbaa1. [](commit_id = 89bbaa130bfdb63242ca9d0aff594947e5ddea8d, deletion_comment = False)

File: tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml (hunk 3 above)
Comment (tiagoshibata) [reply tools/ci_build/github/azure-pipelines/templates/windowsai-nuget-build.yml:47]: No store support in DML yet

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**